operators/api/v1alpha1/tenant_types.go

```diff
@@ -0,0 +1,109 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+See the License for the specific language governing permissions and
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// WorkspaceUserRole is an enum for the role of a user in a workspace
+// +kubebuilder:validation:Enum=Admin;Basic
+type WorkspaceUserRole string
+
+const (
+	// Admin allows to interact with all VMs of a workspace
+	Admin WorkspaceUserRole = "Admin"
+	// Basic allows to interact with owned vms
+	Basic WorkspaceUserRole = "Basic"
+)
+
+// UserWorkspaceData contains the info of the workspaces related to a user
+type UserWorkspaceData struct {
+	WorkspaceURL string            `json:"workspaceURL"`
+	GroupNumber  int               `json:"groupNumber"`
+	Role         WorkspaceUserRole `json:"role"`
+}
+
+// TenantSpec defines the desired state of Tenant
+type TenantSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	Name string `json:"name"`
+
+	SurName string `json:"surname"`
```

Comment:
```suggestion
	Surname string `json:"surname"`
```

Just a proposal though